src/ape/api/providers.py

```diff
@@ -682,18 +682,28 @@ def estimate_gas_cost(self, txn: TransactionAPI, **kwargs) -> int:
 
     @property
     def chain_id(self) -> int:
+        default_chain_id = None
         if self.network.name not in (
             "adhoc",
             LOCAL_NETWORK_NAME,
         ) and not self.network.name.endswith("-fork"):
             # If using a live network, the chain ID is hardcoded.
-            return self.network.chain_id
+            default_chain_id = self.network.chain_id
 
-        elif hasattr(self.web3, "eth"):
-            return self.web3.eth.chain_id
+        try:
+            if hasattr(self.web3, "eth"):
+                return self.web3.eth.chain_id
 
-        else:
-            raise ProviderNotConnectedError()
+        except ProviderNotConnectedError:
+            if default_chain_id:
```

Comment:
I'm assuming there's never a case where `chain_id == 0` right?

Reply:
I changed it to `if default_chain_id is not None` just in case!